### PR TITLE
firefox: add application actions to .desktop file

### PR DESCRIPTION
**Pull request by rhendric**
###### Description of changes

This adds what the Desktop Entry Specification calls application actions for Firefox. Desktop environments present these in various ways (in GNOME, they are available via a right-click on the dash icon). The items added here are those seen in, for example, [Arch](https://github.com/archlinux/svntogit-packages/blob/packages/firefox/trunk/firefox.desktop): New Window and New Private Window.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]
I was actually searching for those today

**Comment by piegamesde** [comment]
Diff looks good to me. However, it got me wondering, the Arch package and Chromium (even on NixOS) have a lot of translations in the desktop file, whereas our Firefox doesn't. Is this something that could be easily added?

**Review by mweinelt — COMMENTED** [review]
I think this is the best maintained upstream desktop file (there are multiple floating around). They also provide a way to enter the profile manager. The casing of all actions is very odd though.

https://github.com/mozilla/gecko-dev/blob/master/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.desktop

**Comment by mweinelt** [comment]
> Is this something that could be easily added?

Well, I hope these don't change very often, so yeah, we could add them. I'd stick with Mozilla as the source for translation strings here.

**Comment by piegamesde** [comment]
@mweinelt Chromium has "New Window" and "New Incognito Window" too. Even the linked desktop file has them, thanks to a weird override: [`Name[en-GB]=New Private Window`](https://github.com/mozilla/gecko-dev/blob/master/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.desktop#L189)

**Comment by mweinelt** [comment]
I have no strong opinion on "Open a [...]" prefix, I just saw that Mozilla was doing it. If that is uncommon we can leave it like that. I have no strong opinions on the matter, given that I don't rely on these actions.

**Comment by mweinelt** [comment]
Very odd indeed.

File: pkgs/applications/networking/browsers/firefox/wrapper.nix [hunk 1]
```diff
@@ -182,6 +184,16 @@ let
...
         startupWMClass = wmClass;
+        actions = {
+          new-window = {
+            name = "New Window";
+            exec = "${launcherName} --new-window %U";
+          };
+          new-private-window = {
+            name = "New Private Window";
+            exec = "${launcherName} --private-window %U";
+          };
```
[the remark this answers is not in the document]
Comment (rhendric) [reply pkgs/applications/networking/browsers/firefox/wrapper.nix:195]: Added, thanks!

**Pull request merged**